### ISSUE TITLE
Rustdoc should not generate docs for hidden methods.

### ISSUE THREAD

**Issue by BenTheElder** [rust-lang/rust#34025]
See previous discussion here: https://github.com/rust-lang/rust/issues/34023#issuecomment-223094149

rustdoc seems to be building files for methods marked `#[doc(hidden)]` in rust 1.9.0 and nightly but not in 1.8.0, observed with code generated when building [opencv-rust](https://github.com/kali/opencv-rust).
It seems that these methods should be ignored.

**Comment by GuillaumeGomez** [comment]
Could you provide a short example on which I could easily please? As far as I can tell, `#[doc(hidden)]` items aren't generated.

**Comment by GuillaumeGomez** [comment]
Also, I made a [patch](https://github.com/GuillaumeGomez/rust/tree/doc_hidden) which is *supposed* to fix this. Can you test it please?

```bash
make check-stage1-rustdocck && cd src/librustdoc/ && ../../x86.../stage1/bin/rustdoc [path of the crate you want to generate] -o [output dir]
```

**Comment by BenTheElder** [comment]
I will take a look at doing both of these in a moment when I get back to a
computer, I think the test case needs to include! the code.

**Comment by GuillaumeGomez** [comment]
If you prefer that I test it, I'll need a short example like I said. So either way, I'm waiting. :wink:

**Comment by BenTheElder** [comment]
So far I haven't managed to trigger it outside of that project, If i get a usable test case I'll post it.
In the mean time I'm also going to start building your patch to test with. Thank you for the quick responses by the way!

**Comment by BenTheElder** [comment]
I think I've got a nice small test case now, I'll post it to github in a moment but the gist is:
```
~/dev/docs_test_case$ tree .
.
├── Cargo.lock
├── Cargo.toml
└── src
    ├── lib.rs
    └── no_doc.rs
```

`lib.rs`:
```
#[doc(hidden)] pub mod sys {
    include!("./no_doc.rs");
}
```

`no_doc.rs`:
```
extern "C" {
#[doc(hidden)] pub fn cv_calib3d_cv_solvePnPRansac_InputArray_objectPoints_InputArray_imagePoints_InputArray_cameraMatrix_InputArray_distCoeffs_OutputArray_rvec_OutputArray_tvec_bool_useExtrinsicGuess_int_iterationsCount_float_reprojectionError_int_minInliersCount_OutputArray_inliers_int_flags();
}
```

**Comment by BenTheElder** [comment]
Test case is here: https://github.com/BenTheElder/docs_test_case

**Comment by GuillaumeGomez** [comment]
Thanks! Testing my patch.

**Comment by GuillaumeGomez** [comment]
It doesn't work. But now I know what to look! :)

**Comment by BenTheElder** [comment]
Of course! Thanks for looking.
The patch was still building for me but I'd be happy to try any future
patches as well.

**Comment by GuillaumeGomez** [comment]
@BenTheElder: Take a look to the PR linked.

**Comment by BenTheElder** [comment]
Thanks, pulling your fix now to test.

**Comment by BenTheElder** [comment]
I *think* the PR works, i did:
```
git checkout doc_hidden
./configure
make check-stage1-rustdocck -j 12
 cd src/librustdoc/
../../x86_64-unknown-linux-gnu/stage1/bin/rustdoc ~/dev/docs_test_case/src/lib.rs
```
Which seems to produce the expected page in `./doc/lib/index.hml` with no methods.

**Comment by GuillaumeGomez** [comment]
Should be fixed by #34045.
